core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -86,7 +89,12 @@ export class ContributionOpportunitiesBackendApiService {
           opportunityType, data.opportunities[index]));
       }
       if (successCallback) {
-        successCallback(opportunities, data.next_cursor, data.more);
```

Comment:
Why is this change being done in this PR?

Reply:
The `.then()` method returns `value?: Object | PromiseLike<Object>` and I can't change this. The only workaround I saw was to turn these params into a single object and change the effected lines of code

Reply:
I'm not against the change -- it is good to have one dict going through successCallback.

Though, I'm curious `.then` was accepting all the three parameters before, right? `.then((opportunities, nextCursor, more)` why it is accepting just one object now? Is it the case that in Angular it behaves this way and in AngularJS it behaves the other way?

Reply:
@bansalnitish To be honest, I had no idea why but when I checked the type definition for `Promise.then`:
<img width="896" alt="Screenshot 2020-05-04 at 13 32 53" src="https://user-images.githubusercontent.com/11369764/80961702-c6d73b00-8e0b-11ea-9f95-94efe011fda5.png">

Reply:
@bansalnitish, is this thread resolved?

Reply:
@h-ARTS could you just try this one locally and let us know if we could pass the three params too? (I think it would be good to know for you as well)

Reply:
@h-ARTS, Can you please resolve this comment? [Make sure to verify all the comments are resolved!] Thanks!

Reply:
@DubeySandeep @bansalnitish I could resolve this one. This way I can get rid of these interfaces:
```ts
export interface IFetchedSkillOpportunitiesSuccessCallbackParams {
  opportunities: SkillOpportunity[],
  nextCursor: string,
  more: boolean
}

export interface IFetchedExplorationOpportunitiesSuccessCallbackParams {
  opportunities: ExplorationOpportunitySummary[],
  nextCursor: string,
  more: boolean
}
```

**However** I need to specify `any` as the return in these methods in the backend-api service:
```ts
// TODO(#7165): Replace 'any' with the exact type.
  // This has been typed as 'any' because 'resolve' returns
  // a Promise: (value: T | PromiseLike<T>): Promise<T>
  fetchSkillOpportunities(cursor: string): any {
    const params: ContributionOpportunityParams = {
      cursor: cursor
    };
    return new Promise((resolve, reject) => {
      this._fetchOpportunities(
        constants.OPPORTUNITY_TYPE_SKILL, params, resolve, reject);
    });
  }

  // TODO(#7165): Replace 'any' with the exact type.
  // This has been typed as 'any' because 'resolve' returns
  // a Promise: (value: T | PromiseLike<T>): Promise<T>
  fetchTranslationOpportunities(languageCode: string, cursor: string): any {
    const params: ContributionOpportunityParams = {
      language_code: languageCode,
      cursor: cursor
    };
    return new Promise((resolve, reject) => {
      this._fetchOpportunities(
        constants.OPPORTUNITY_TYPE_TRANSLATION,
        params, resolve, reject);
    });
  }

  // TODO(#7165): Replace 'any' with the exact type.
  // This has been typed as 'any' because 'resolve' returns
  // a Promise: (value: T | PromiseLike<T>): Promise<T>
  fetchVoiceoverOpportunities(languageCode: string,
      cursor: string): any {
    const params: ContributionOpportunityParams = {
      language_code: languageCode,
      cursor: cursor
    };
    return new Promise((resolve, reject) => {
      this._fetchOpportunities(
        constants.OPPORTUNITY_TYPE_VOICEOVER,
        params, resolve, reject);
    });
  }
```